### ISSUE TITLE
Need to add Tag field to simple, producer-only order cycle interface

### ISSUE THREAD

**Issue by sstead** [openfoodfoundation/openfoodnetwork#2140]
## Description
We have the ability to setup tag rules showing/hiding OCs. However this can't be done for a producer only OC, because there is no field to type the tag. In a hub OC you see the tag field in the outgoing section, but this isn't there for simple OCs.

## Steps to Reproduce
1. Look at any producer only OC, you'll notice there's no tag field

## Animated Gif/Screenshot
![image](https://user-images.githubusercontent.com/10843056/37382928-b4fcc894-2799-11e8-9cb2-13b251816a67.png)

## Context
I haven't had a user request this yet, but it's part of our core functionality, and it should be a simple fix.

## Severity
3 - not too many users are impacted, but it is core functionality

## Possible Fix
Add tag field to simple OCs

**Comment by sstead** [comment]
Note: TFG have encountered this bug in Australia - they now have to make a work around on top of a work around... Because using tagging itself is a work around, and now they have to become a 'hub' just to use the work around...annoying

**Comment by sstead** [comment]
@myriamboure or anyone, let me know if this is not in fact a good first issue (seems straightforward to me, but I'm no dev).

**Comment by daniellemoorhead** [comment]
I've added this to the delivery train backlog @sstead, would be great to get an occasional contributor to pick it up given it's not too complex (pending a developer saying as much..ping @oeoeaio for confirmation on this).

**Comment by oeoeaio** [comment]
Hm, the order cycles interface is notoriously complex and angular-heavy, but perhaps the producer-only interface is not so crazy. I am happy to have a look at this once I've finished subs work, or to provide direction to others if required.

**Comment by myriamboure** [comment]
I added a mockup to precise how this would look like, @sstead tell me if that's how you see it as well. If yes maybe we can ask Luis to work on it he is doing more complex things, unless Rob is one with subscriptions and want to do it of course :-)

**Comment by daniellemoorhead** [comment]
@myriamboure I've assigned to @luisramos0 and moved to dev ready.

@luisramos0 it's so good to have you onboard and contributing 😃

**Comment by luisramos0** [comment]
Unassigning myself for now. I am not planning to pick this one up. Spree upgrade...

**Comment by RachL** [comment]
@daniellemoorhead @myriamboure this looks as a nex feature for me not a bug. I would like to propose to move this to discourse. Do you agree?

**Comment by daniellemoorhead** [comment]
I'm not sure @RachL, I guess it can be argued that it was designed that way and therefore not a bug...but it also doesn't feel right to turn something so small into a wishlist item.

@lin-d-hop what do you think? You'll be looking after the Admin UX work, does this feel like something that would go into that uber list of things?

**Comment by luisramos0** [comment]
Let's not assume this is small.
I don't think we should consider a request for a new UI component a bug... it would be kind of a pandora box if we do that, we have so many UX inconsistencies in the app...

**Comment by RachL** [comment]
I would like to add that this is not a UX issue. This is bringing an existing feature available to another group of users. 

- On a side note, I didn't know we were starting to build kind of a UX admin work/epic. @lin-d-hop how can I help / contribute on that?

**Comment by luisramos0** [comment]
yeah, that's it, it's about making the UI consistent across user profiles. from this perspective this is a feature, isnt it?

**Comment by Erioldoesdesign** [comment]
hmm I'm finding this one a bit hard to decide on...i don't see a problem with adding an existing functionality (tags) from hub users over to producer only users but I do have a problem with adding new UI layout in the mockup in the issue. I would say if this is to be done it should look like how it looks in the hub user case, inside the table in outgoing that you expand.

In general...tags are confusing unless you know exactly how to use them and adding them in the 'top level' like the mock up without any links to docs, explanation text etc. is not the best UX.

**Comment by luisramos0** [comment]
Hi @Erioldoesdesign 
The pages are quite different, here we dont have 3 pages (settings/incoming/outgoing) and we dont have those expandable tables as seen in the second screenshot in the description above. Here we only have this simple product list as seen in the first screenshot. So, it's not clear what do you mean by "it should look like how it looks in the hub user case"

**Comment by luisramos0** [comment]
simple: https://staging.coopcircuits.fr/admin/order_cycles/98/edit
outgoing of complex: https://staging.coopcircuits.fr/admin/order_cycles/2/outgoing

**Comment by Erioldoesdesign** [comment]
Again, perhaps this is just me not clearly understanding tags and why they exist and for what purpose from a user perspective. I'm sure many people internally can tell me what the intention of tags were.

My understanding of tags so far is they are used in combination with order cycles for subscriptions:
<img width="1049" alt="Screenshot 2020-11-16 at 12 56 38" src="https://user-images.githubusercontent.com/11681324/99258306-46f54900-2810-11eb-8f91-babcc23977ba.png">

and also in other cases such as members of a shop or maybe special shipping/payment for a specific case e.g. 'Christmas order' tag: 'xmasdelivery' or something like that.

Though as I was playing in the back office I was able to add 'junk tags' which was very confusing:
<img width="1162" alt="Screenshot 2020-11-16 at 12 58 30" src="https://user-images.githubusercontent.com/11681324/99259064-62ad1f00-2811-11eb-8922-1166829328a8.png">


My problem is not so much the paper cut suggestion here and it's validity as a papercut for tech, it's whether it makes sense for users of the tag feature/function overall is where I'm skeptical.
Sure it's easy to move a function like tagging from one area to a more 'global' area so that tags are easier to find and work on simple OC's and complex OC's but does that make tagging make more sense to use or not? Are we applying a logical user process/journey to the act of tagging? I would say no.

But it is likely this makes sense for users that understand tagging within the complexity of OFN already.

**Comment by RachL** [comment]
@Erioldoesdesign some things that might help:

> order cycles for subscriptions:

Those are not tags.
They are schedules, both are not linked.

> I was playing in the back office I was able to add 'junk tags'

Yes I understand it can be confusing, but it is also nice to be able to leave flexibility here for now, while the whole tag journey is not redesigned.
To make a tag work you need especially a tag rule, have you checked https://guide.openfoodnetwork.org/basic-features/shopfront/customer-management-and-conditional-displays-prices/tags-and-tag-rules ? 

Re: do simple profile users needs this? I think the general answer is yes as no matter how complex tags are they are way more flexible and less difficult to master than inventory. 
So currently what happens is that we switch simple profile users to hubs so they can have access to the tag feature within OC. But this means they also get the full OC version with many other features they don't need. Some of them will make them loose time and create mistakes (e.g. having to repeat their action on step 2 and step 3). This is I believe what this papercut is trying to solve.
